lib/src/ardrive_io.dart

```diff
@@ -36,4 +48,6 @@ abstract class ArDriveIO {
   Future<IOFolder> pickFolder();
 
   Future<void> saveFile(IOFile file);
+
+  Stream<SaveStatus> saveFileStream(IOFile file, Completer<bool> finalize);
```

Comment:
Consider adding a new abstraction to define this method. For instance, 

```dart
abstract class StreamFileSaver {
  Stream<SaveStatus> saveFileStream(IOFile file, Completer<bool> finalize);
}
```
and, on the file `DartIOFileSaver` you can:

```dart
class DartIOFileSaver implements StreamFileSaver {}
```

So, we segregate the interfaces for components that don't use/need. What do you think?

Reply:
Would we be keeping the `saveFileStream` method on `ArDriveIO`? If so then what should happen when using `FileSaver`s that don't have that `StreamFileSaver` interface?